src/main/java/com/firebolt/jdbc/type/JavaTypeToFireboltSQLString.java

```diff
@@ -24,23 +28,53 @@
 
 public enum JavaTypeToFireboltSQLString {
 	BOOLEAN(Boolean.class, value -> Boolean.TRUE.equals(value) ? "1" : "0"),
-	UUID(java.util.UUID.class, value -> ((UUID) value).toString()),
-	SHORT(Short.class, value -> Short.toString((short) value)),
+	UUID(java.util.UUID.class, Object::toString),
+	SHORT(Short.class, value -> Short.toString(((Number) value).shortValue())),
 	STRING(String.class, getSQLStringValueOfString()),
-	LONG(Long.class, String::valueOf),
-	INTEGER(Integer.class, String::valueOf),
-	BIG_INTEGER(BigInteger.class, String::valueOf),
-	FLOAT(Float.class, String::valueOf),
-	DOUBLE(Double.class, String::valueOf),
+	LONG(Long.class, value -> Long.toString(((Number)value).longValue())),
+	INTEGER(Integer.class, value -> Integer.toString(((Number)value).intValue())),
+	BIG_INTEGER(BigInteger.class, value -> value instanceof BigInteger ? value.toString() : Long.toString(((Number)value).longValue())),
+	FLOAT(Float.class, value -> Float.toString(((Number)value).floatValue())),
```

Comment:
Can you explain why did we switch from `valueOf`? What's the advantage here?

Reply:
ValueOf just calls `toString()`. I wanted to be sure that the passed value (of type Object) indeed can be transformed to the required type. This never happens if you use typed setters like setInt() or setDouble() but can happen when you use setObject(). For example one can say

```java
ps.setObject(1, "one", Types.INTEGER); 
```

The compilation will obviously work but we want to fail at runtime in this line and not wait for the statement execution.